.circleci/config.yml

```diff
@@ -216,60 +216,20 @@ commands:
           keys:
             - 'libsodium-fork-{{ .Environment.CIRCLE_STAGE }}-{{ checksum "tmp/libsodium.md5" }}'
 
-  generic_shorttest:
-    description: Run short tests from build workspace, for re-use by diferent architectures
+  generic_buildtest:
+    description: Run build tests from build workspace, for re-use by diferent architectures
     parameters:
       circleci_home:
         type: string
         default: "/home/circleci"
       result_subdir:
         type: string
-    steps:
-      - attach_workspace:
-          at: << parameters.circleci_home >>
-      - run: mkdir -p /tmp/results/<< parameters.result_subdir >>
-      - restore_cache:
-          keys:
-            - 'go-cache-{{ .Environment.CIRCLE_STAGE }}-'
-      - run:
-          name: Run short tests
-          no_output_timeout: 30m
-          command: |
-            set -e
-            set -x
-            export PATH=$(echo "$PATH" | sed -e 's|:<< parameters.circleci_home >>/\.go_workspace/bin||g' | sed -e 's|:/usr/local/go/bin||g')
-            export KMD_NOUSB=True
-            export GOPATH="<< parameters.circleci_home >>/go"
-            export PATH="${PATH}:${GOPATH}/bin"
-            export ALGORAND_DEADLOCK=enable
-            GOLANG_VERSION=$(./scripts/get_golang_version.sh)
-            eval "$(~/gimme "${GOLANG_VERSION}")"
-            scripts/configure_dev.sh
-            scripts/buildtools/install_buildtools.sh -o "gotest.tools/gotestsum"
-            PACKAGES="$(go list ./... | grep -v /go-algorand/test/)"
-            export PACKAGE_NAMES=$(echo $PACKAGES | tr -d '\n')
-            export PARTITION_TOTAL=$CIRCLE_NODE_TOTAL
-            export PARTITION_ID=$CIRCLE_NODE_INDEX
-            export PARALLEL_FLAG="-p 1"
-            gotestsum --format pkgname --junitfile /tmp/results/<< parameters.result_subdir >>/results.xml --jsonfile /tmp/results/<< parameters.result_subdir >>/testresults.json -- --tags "sqlite_unlock_notify sqlite_omit_load_extension" -short -race -timeout 1h -coverprofile=coverage.txt -covermode=atomic -p 1 $PACKAGE_NAMES
-      - store_artifacts:
-          path: /tmp/results
-          destination: test-results
-      - store_test_results:
-          path: /tmp/results
-      - save_cache:
-          key: 'go-cache-{{ .Environment.CIRCLE_STAGE }}-{{ .Environment.CIRCLE_BUILD_NUM }}'
-          paths:
-            - tmp/go-cache
-
-  generic_nightlytest:
-    description: Run nightly tests from build workspace, for re-use by diferent architectures
-    parameters:
-      circleci_home:
+      no_output_timeout:
         type: string
-        default: "/home/circleci"
-      result_subdir:
+        default: 30m
+      short_test_flag:
```

Comment:
I wonder if this could be made more generic to support any test flag?

Reply:
I'm not sure I want to do that anymore:
1. We're calling a bunch of other parameters inside the test itself right now, so it could be confusing to add an any_test flag because you can't add whatever parameter you want. Only parameters that haven't been specified yet
2. `-short` is the main flag we're using and it helps to repeat the same variable name used in generic_integration for consistency
3. Will mentioned that he wants us to stop calling gotestsum directly in the file eventually, so I expect further modifications on this command. (#1426 in our CircleCI epic)